### PR TITLE
04/11/2016

### PR DESCRIPTION
**Pull request by avbel**
Added first demo - phone number verification.
Rest to write tests

### CODE REVIEW

File: src/demos/phoneNumberVerificator.js [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+import debugFactory from 'debug';
+import fetch, {Headers} from 'node-fetch';
+import {PhoneVerificationLogEntry} from '../models';
+
+const debug = debugFactory('phoneNumberVerificator');
+
+export default function () {
+	const router = new Router();
+
+	router.post('/sendCode', getServicePhoneNumber, async context => {
```
Comment (dtolb) [inline src/demos/phoneNumberVerificator.js]: I like this pattern! Smart :)

**Comment by dtolb** [comment]
Any reason package.json is removed/replaced each time? Just curious.

File: src/demos/phoneNumberVerificator.js [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+<Response>
+	<SpeakSentence locale="en_US" gender="female" voice="julie">Your confirm code is ${code}</SpeakSentence>
+	<Hangup/>
+</Response>`;
+	});
+
+	return router;
+}
+
+async function getServicePhoneNumber(context, next) {
```
Comment (dtolb) [inline src/demos/phoneNumberVerificator.js]: Can you break this function out to a 'common.js' or something. I figure we'll be doing this check a lot

File: src/demos/phoneNumberVerificator.js [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+		entry.verifiedTime = new Date();
+		await entry.save();
+		context.body = '';
+		return;
+	});
+
+	router.get('/callCallback', async context => {
+		const code = context.query.tag;
+		debug('Catapult callback query %j', context.query);
+		if (!code || context.query.eventType !== 'answer') {
```
Comment (dtolb) [inline src/demos/phoneNumberVerificator.js]: Can you also add this functionality: "Speak sentence in call" to a new function. We'll probably use this a lot as well

File: src/demos/phoneNumberVerificator.js [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+	});
+
+	return router;
+}
+
+async function getServicePhoneNumber(context, next) {
+	console.log('Test');
+	if (!context.session.servicePhoneNumber) {
+		// replace this code by lib's call when it will be implemented
+		const auth = context.user.catapultAuthData;
```
Comment (dtolb) [inline src/demos/phoneNumberVerificator.js]: Can break down to a separate function again.

File: src/demos/phoneNumberVerificator.js [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+		if (numbers && numbers[0]) {
+			context.session.servicePhoneNumber = numbers[0].number;
+		} else {
+			headers.set('Content-Type', 'application/json');
+			const numberRequest = {
+				city: 'Cary',
+				state: 'NC',
+				quantity: 1
+			};
+			response = await fetch(`https://api.catapult.inetwork.com/availableNumbers/local`, {method: 'POST', headers, body: JSON.stringify(numberRequest)}).then(checkStatus);
```
Comment (dtolb) [inline src/demos/phoneNumberVerificator.js]: New function again :smile:

**Comment by dtolb** [comment]
This is looking really good! 

Overall, try to abstract any catapult interactions to as generic as possible functions.  Ideally someone new to catapult could see at a high level how we're interacting with catapult. 

When I tried to run the verify number locally, I got invalid credentials on my creds. I made sure I was using the right ones but I could have made a mistake. Can you double check and make sure that's tight.

We also have a new theme to apply: https://github.com/inetCatapult/bandwidth-page-template/tree/master/New (let me know if you don't have access to that repo). We should use this theme here. We can chat about that a bit more tomorrow :smile:

Keep up the good work!

File: src/demos/phoneNumberVerificator.js (hunk 2 above)
Comment (avbel) [inline src/demos/phoneNumberVerificator.js]: ok

File: src/demos/phoneNumberVerificator.js (hunk 3 above)
Comment (avbel) [inline src/demos/phoneNumberVerificator.js]: ok. no problem :)

File: src/demos/phoneNumberVerificator.js (hunk 4 above)
Comment (avbel) [inline src/demos/phoneNumberVerificator.js]: sure

File: src/demos/phoneNumberVerificator.js (hunk 5 above)
Comment (avbel) [inline src/demos/phoneNumberVerificator.js]: :)

**Pull request closed without merging**